internal/cli/install/helm.go

```diff
@@ -0,0 +1,311 @@
+	"helm.sh/helm/v3/pkg/cli"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	tools "capact.io/capact/internal"
+	"capact.io/capact/internal/helmhelper"
+	"capact.io/capact/internal/ptr"
+)
+
+type chartInstallFunc func(version string) error
+
+func getActionConfiguration(k8sCfg *rest.Config, forNamespace string) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
+	helmCfg := &genericclioptions.ConfigFlags{
+		APIServer:   &k8sCfg.Host,
+		Insecure:    &k8sCfg.Insecure,
+		CAFile:      &k8sCfg.CAFile,
+		BearerToken: &k8sCfg.BearerToken,
+		Namespace:   ptr.String(forNamespace),
+	}
+
+	debugLog := func(format string, v ...interface{}) {
+		// noop
+	}
+
+	err := actionConfig.Init(helmCfg, forNamespace, "secrets", debugLog)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "while initializing Helm configuration")
+	}
+
+	return actionConfig, nil
+}
+
+type Helm struct {
+	configuration *action.Configuration
+	opts          Options
+}
+
+func NewHelm(configuration *action.Configuration, opts Options) *Helm {
+	if opts.Parameters.IncreaseResourceLimits {
+		opts.Parameters.Override.CapactValues.Gateway.Resources = upgrade.IncreasedGatewayResources()
+		opts.Parameters.Override.CapactValues.HubPublic.Resources = upgrade.IncreasedHubPublicResources()
+		opts.Parameters.Override.Neo4jValues.Neo4j.Core.Resources = upgrade.IncreasedNeo4jResources()
+	}
+	if opts.Parameters.Override.HelmRepoURL == capact.HelmRepoLatestTag {
+		opts.Parameters.Override.HelmRepoURL = capact.HelmRepoLatest
+	}
+
+	return &Helm{configuration: configuration, opts: opts}
+}
+
+func (h *Helm) InstallCharts(status *printer.Status) error {
+	var charts = map[Chart]chartInstallFunc{
+		neo4j:             h.installNeo4j,
+		ingressController: h.installIngressController,
+		argo:              h.installArgo,
+		certManager:       h.installCertManager,
+		kubed:             h.installKubed,
+		monitoring:        h.installMonitoring,
+		capactChart:       h.installCapact,
+	}
+
+	var err error
+	helper := helmhelper.NewHelmHelper()
+
+	for _, chart := range chartsOrder {
+		if shouldSkipTheChart(chart, h.opts.SkipCharts) {
+			fmt.Println("skipping chart", chart)
+			continue
+		}
+
+		version := h.opts.Parameters.Version
+		if version == capact.LatestVersionTag {
+			version, err = helper.GetLatestVersion(h.opts.Parameters.Override.HelmRepoURL, string(chart))
+			if err != nil {
+				return errors.Wrapf(err, "while getting latest version for %s", chart)
+			}
+		}
+		installFunc, ok := charts[chart]
+		if !ok {
+			return fmt.Errorf("could not find install function for %s", chart)
+		}
+
+		status.Step("Installing %s Helm Charts", chart)
+		err = installFunc(version)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (h *Helm) installNeo4j(version string) error {
```

Comment:
In the next PR I will split it so each Component will be a separate object:

Something like
```go
type Component interface {
  func Install(version string) error
  func Uninstall() error
  func Build() error
}
```